### PR TITLE
Feat/zap automation

### PR DESCRIPTION
**Pull request by jusdino**
[#zap](https://github.com/csg-org/CompactConnect/issues/227)

### CODE REVIEW

File: owasp-zap/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+[ZAP](https://www.zaproxy.org/) is a security pen testing tool, which this project uses for active scanning for
+vulnerabilities. To manually run a scan similar to the ones integrated into [our GitHub Actions](../.github/workflows),
+run `./manual-scan.sh` from a command line. For full details on running the scan locally, see [Manual run ](#manual-run). Below is a brief explanation of the pieces of this integration and what they are for.
+
+# Authenticator
+
+The [authenticator](./authenticator) directory is a simple NodeJS script that leverages
+[aws-amplify](https://www.npmjs.com/package/aws-amplify) to quickly log a test user in and acquire an token for
+authentication. This script requires that the Cognit UserPool Client be configured to enable the SRP authentication
```
Comment (landonshumway-ia) [inline owasp-zap/README.md:11]:
```suggestion
authentication. This script requires that the Cognito UserPool Client be configured to enable the SRP authentication
```

File: owasp-zap/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+flow, which can only be activated in pre-production environments.
```
Comment (landonshumway-ia) [inline owasp-zap/README.md:12]: Can you explain a bit more here on why we can only activate it in pre-prod? Is this related to MFA requirements?

File: owasp-zap/README.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+# Data
+
+The [data](./data) folder contains configuration and automation data files that are used to control ZAP's scan. This
+includes an HttpSender script and an automation YML file. The HTTPSender script will tell ZAP to add the token we acquired from the [authenticator](#authenticator) script into the `Authorization` header for every 'in scope' request, headed
+to the UI or API. The YML file defines what jobs are to be run as part of an automated scan.
+
```
Comment (landonshumway-ia) [inline owasp-zap/README.md:19]: Do we need a section here for how to update to the ZAP data in the event that more API endpoints are added or other resources need to be tested?

File: owasp-zap/authenticator/Gruntfile.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
+//
+//  Gruntfile.js
+//  CompactConnect
+//
+//  Created by InspiringApps on 7/22/2024.
```
Comment (landonshumway-ia) [inline owasp-zap/authenticator/Gruntfile.js:5]: Do we want to have a reference here to InpsiringApps? My understanding was we usually don't want the company name anywhere in the code base.

**Review by landonshumway-ia — CHANGES_REQUESTED** [review]
The code looks good, of course I don't know much about ZAP. Nice work on refactoring the tests. Just found a couple things in the new README worth looking at.

File: owasp-zap/README.md (hunk 3 above)
Comment (jusdino) [reply owasp-zap/README.md:19]: No need. This ZAP automation includes ingesting our official API Spec via latest-oas30.json, so it will always have the latest API definition as part of its input. Also, we're going to add similar automation for practitioner users as a second scan, probably, next.

File: owasp-zap/authenticator/Gruntfile.js (hunk 4 above)
Comment (jusdino) [reply owasp-zap/authenticator/Gruntfile.js:5]: That's following the example set in `webroot`, but I agree - I'll remove it.

**Review by landonshumway-ia — APPROVED** [review]
🚀
